routing/router.go

```diff
@@ -1142,6 +1140,15 @@ func (r *ChannelRouter) processUpdate(msg interface{}) error {
 
 		}
 
+		// If the channel doesn't exist in our database, we cannot
+		// apply the updated policy.
+		if !exists {
+			return newErrf(ErrIgnored, "Ignoring update "+
```

Comment:
We return a slightly different error object than before. Is it okay with the callers?

Reply:
Can the situation occur if something happens in between the two db queries for the channel?

Reply:
Even if it does, the error is handled correctly by the caller here:
https://github.com/lightningnetwork/lnd/blob/351865ac6772e7885a64c7e0c1e154abade40c91/discovery/gossiper.go#L2034

Reply:
Ok, clear.

The other use of `UpdateEdge` is in `ChannelRouter.applyChannelUpdate`. But I think the behaviour improves there too?

Reply:
Yes looks okay: https://github.com/lightningnetwork/lnd/blob/e10fe91f10efe46c540456ee5e22bfbd929fbe0d/routing/router.go#L2054